docs/code-search/code-navigation/auto_indexing.mdx

```diff
@@ -0,0 +1,320 @@
+
+### Enable index job scheduling
+
+Next, enable the precise code navigation auto-indexing feature by enabling the following feature flag in your Sourcegraph instance's site configuration.
+
+```yaml
+{
+  "codeIntelAutoIndexing.enabled": true
+}
+```
+
+This step will control the scheduling of indexing jobs which are made available to the executors deployed in the previous step.
+
+### Configure auto-indexing policies
+
+Once auto-indexing has been enabled, [create auto-indexing policies](/code_navigation/how-to/configure_auto_indexing) to control the set of repositories and commits that are eligible for indexing. Note that policies only select repositories and branches/tags/commits that can be indexed, not how they are indexed.
+
+### Configure auto-indexing jobs
+
+For repository and commit pairs that are marked as eligible for indexing, the index job is either [inferred from the project](/code_navigation/explanations/auto_indexing_inference), or [explicitly configured](/code_navigation/how-to/configure_auto_indexing#explicit-index-job-configuration) on a per-repository basis.
+
+### Tune the index scheduler
+
+The frequency of index job scheduling can be tuned via the following environment variables read by `worker` service containers running the [`codeintel-auto-indexing`](/admin/workers#codeintel-auto-indexing) task.
+
+- **`PRECISE_CODE_INTEL_AUTO_INDEXING_TASK_INTERVAL`**: The frequency with which to run periodic codeintel auto-indexing tasks. Default is every 2 minutes
+- **`PRECISE_CODE_INTEL_AUTO_INDEXING_REPOSITORY_PROCESS_DELAY`**: The minimum frequency that the same repository can be considered for auto-index scheduling. Default is every 24 hours
+- **`PRECISE_CODE_INTEL_AUTO_INDEXING_REPOSITORY_BATCH_SIZE`**: The number of repositories to consider for auto-indexing scheduling at a time. Default is 2500
+- **`PRECISE_CODE_INTEL_AUTO_INDEX_MAXIMUM_REPOSITORIES_INSPECTED_PER_SECOND`**: The maximum number of repositories inspected for auto-indexing per second. Set to zero to disable limit. Default is 0
+
+### Access to private repositories and packages
+
+Auto-indexing jobs run as Docker containers on the Executors infrastructure, and by default don't have any knowledge about the private repositories or package registries your builds might use.
+
+The type of information you will need to provide depends on the language and ecosystem, but the mechanism is the same - [adding a secret to Executors "Code Graph" namespace](/admin/executors/executor_secrets).
+
+See the [language-specific instructions](/code_navigation/how-to/ configure_auto_indexing#private-repositories-and-packages-configuration)
+
+## Configure Code Navigation auto-indexing
+
+<Callout type="info">This feature is in beta for self-hosted customers.</Callout>
+
+Precise code navigation [auto-indexing](/code_navigation/explanations/auto_indexing) jobs are scheduled based on two fronts of configuration.
```

Comment:
The inconsistent use of 'auto-indexing' vs 'Code navigation auto-indexing' vs 'Precise code navigation auto-indexing' is quite distracting. Also, we will soon be adding a new form of auto-indexing, which will not be precise, so we have been calling it 'syntactic auto-indexing' internally.

For that, _most_ of the settings will be shared with precise auto-indexing, with some minor tweaks. So I propose simplifying the terminology to only use 'auto-indexing' for now. For the future, in places we need a finer distinction, we can use 'precise auto-indexing' and 'syntactic auto-indexing' as shorthand for 'auto-indexing for precise code navigation' and 'auto-indexing for syntactic code navigation respectively'.

Reply:
Nice callout and I agree here. Let's simplify it, and maybe we can change it later.